src/ErrorBoundaries/ErrorBoundaries.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react"
+
+import { ErrorFallbackComponent } from "./ErrorFallbackComponent"
+
+type Props = {
+  children: JSX.Element
+  onError?: () => void
+}
+
+type State = { error: Error | string; hasError: boolean }
+
+export class ErrorBoundary extends React.Component<Props, State> {
```

Comment:
What do you think about using a functional component here?

Reply:
or do errorboundaries still need to be extended from React.Component?

Reply:
Yeah, unfortunately, we still don't have any hooks to support error boundaries.